Source/Initialization/InjectorMomentum.H

```diff
@@ -86,6 +87,66 @@ private:
     amrex::Real m_ux_th, m_uy_th, m_uz_th;
 };
 
+namespace {
+    /** Return u sampled according to the probability distribution:
+      * p(u) \propto u \exp(-(u-u_m)^2/2u_th^2)
+      *
+      * @param u_m Central momentum
+      * @param u_th Momentum spread
+      * @param engine Object used to generate random numbers
+      */
+    AMREX_GPU_HOST_DEVICE
+    amrex::Real
+    generateGaussianFluxDist( amrex::Real u_m, amrex::Real u_th, amrex::RandomEngine const& engine ) {
+
+        using namespace amrex::literals;
+
+        // Momentum to be returned at the end of this function
+        amrex::Real u = 0._rt;
+
+        if (u_th == 0._rt) {
+            u = u_m; // Trivial case ; avoids division by 0 in the rest of the code below
+        } else if (u_m < 0.6*u_th) {
+            // Mean velocity is lower than thermal velocity
+            // Use the distribution u*exp(-u**2*(1-u_m/u_th)/(2*u_th**2)) as an approximation
+            // and then use the rejection method to correct it
+            // ( stop rejecting with probability exp(-u_m/(2*u_th**3)*(u-u_th)**2) )
```

Comment:
Maybe the math is clear enough, but perhaps a comment could be added that when `u_m = 0` the first guess is always used (so that this doesn't add any inefficiency for that common case).

Reply:
OK, I will add this.